### PR TITLE
fix(TDI-40112): tCloudStart doesn't start an existing EC2 instance

### PR DESCRIPTION
**Pull request by undx**
**What is the current behavior?** (You can also link to an open issue here)

tCloudStart does not start an existing EC2 instance but creates a new one.
https://jira.talendforge.org/browse/TDI-40110
https://jira.talendforge.org/browse/TDI-40112

**What is the new behavior?**
If the requested instance count is set to 1, we should 
* start this instance if it's suspended.
* do nothing if it's already running.
* create a new one if it's terminated (deleted). The newly created instance will use the `instanceName` as base name.

Otherwise, we'll create _n_ instances using the `instanceName` as base name.

After merge, need to be backported to  `maintenance/6.4` and `maintenance/6.5`.

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tCloudStart/tCloudStart_begin.javajet [hunk 1]
```diff
@@ -65,76 +65,122 @@
...
+        nodes_<%=cid%> = client_<%=cid%>.listNodesDetailsMatching(new com.google.common.base.Predicate<org.jclouds.compute.domain.NodeMetadata>() {
+             @Override
+             public boolean apply(org.jclouds.compute.domain.NodeMetadata nodeMetadata) {
+                 if (<%=instanceName%>.equals(nodeMetadata.getName())) {
+                     return true;
+                 }
+                 return false;
+             }
+        });
+        String instanceId = null;
```
Comment (ypiel-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tCloudStart/tCloudStart_begin.javajet]: No need to have instanceId_<%=cid %> suffixe ?

File: main/plugins/org.talend.designer.components.localprovider/components/tCloudStart/tCloudStart_begin.javajet [hunk 2]
```diff
@@ -65,76 +65,122 @@
...
+            System.out.println("Starting " + <%=instanceName%> + " (" + instanceId + ")...");
+            try {
+                client_<%=cid%>.resumeNode(instanceId);
+                hasToCreateNode_<%=cid%> = false;
+           	} catch(com.google.common.util.concurrent.UncheckedExecutionException e_<%=cid%>) {
+           		boolean ignoreException_<%=cid%> = false;
+           		if(e_<%=cid%>.getCause()!=null && (e_<%=cid%>.getCause() instanceof java.lang.NullPointerException)) {
+           			if("name".equals(e_<%=cid%>.getCause().getMessage())) {
+           				ignoreException_<%=cid%> = true;
+           				System.err.println("Some exception happen when get the running instance information object, you can avoid it by setting the key pair");
```
Comment (ypiel-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tCloudStart/tCloudStart_begin.javajet]: syserr not sl4j ?

Comment (undx) [reply main/plugins/org.talend.designer.components.localprovider/components/tCloudStart/tCloudStart_begin.javajet]: Was a copy/paste from exception management below.

File: main/plugins/org.talend.designer.components.localprovider/components/tCloudStart/tCloudStart_begin.javajet (hunk 1 above)
Comment (undx) [reply main/plugins/org.talend.designer.components.localprovider/components/tCloudStart/tCloudStart_begin.javajet]: Yes, not sure needed.

File: main/plugins/org.talend.designer.components.localprovider/components/tCloudStart/tCloudStart_begin.javajet (hunk 2 above)
Comment (undx) [reply main/plugins/org.talend.designer.components.localprovider/components/tCloudStart/tCloudStart_begin.javajet]: Reworking the logging.

**Pull request merged**